### ISSUE TITLE
Decide the meaning of the return codes, specify it and update the code accordingly

### ISSUE THREAD

**Issue by Niols** [colis-anr/colis-language#32]

**Comment by Niols** [comment]
Proposition:

| Code | Meaning |
|---------|--------|
| 0 | Success of the execution ending up on a success |
| 1 | Success of the execution ending up on a failure |
| 2 | Error in command-line parsing |
| 3 | Error in parsing (Shell or CoLiS) |
| 4 | Error in conversion |
| 5 | Error in pretty-printing (Shell or CoLiS) |
| 6 | Error in execution (concrete or symbolic) |
| ? | Unexpected error |

We might have to add some new error codes, so I'd say that the "unexpected error" code should be high, like 125 or 99, or whatever.

If we agree on such a table, this should go in the documentation. Then, they whole tool should be modified according to this convention

**Comment by benozol** [comment]
Good idea. There is also `Sys_error` when opening a script file (i.e., file not found). Or should this just return `2`?

**Comment by Niols** [comment]
Hum, it does not cost much to have a separate return code for this. And I like to have a difference between "Error in command-line parsing" (the user made a mistake in its use of the tool) and "File not found" (which can be a user mistake, or not)

**Comment by Niols** [comment]
7 for unsupported utility or argument?

**Comment by Niols** [comment]
We should probably let 2 to OCaml's uncaught exceptions

**Comment by Niols** [comment]
Updated proposition:

Code | Meaning
-- | --
0 | Success of the execution ending up on a success
1 | Success of the execution ending up on a failure
2 | Unexpected OCaml exception
3 | Error in command-line parsing (user's mistake)
4 | File does not exist (might be during cmd parsing as well)
5 | Error in parsing (Shell or CoLiS)
6 | Unsupported structure (Error in conversion)
7 | Unsupported utility
8 | Unsupported option for a supported utility
9 | Error in pretty-printing (Shell or CoLiS)
… | …
125 | Unexpected error

Any comments?
